build/Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM alpine:3.6
```

Comment:
I noticed that the current SDK generates a Dockerfile using ubi-minimal ([RedHat UBI](https://www.redhat.com/en/blog/introducing-red-hat-universal-base-image)) by default. The original [motivation](https://github.com/operator-framework/operator-sdk/pull/1139) for this was apparently to make it easier to deploy on OpenShift -- although I'm not sure exactly why using UBI helps; maybe it's just a matter of support.

We should consider the pros and cons of moving to using UBI here. I think we will be using UBI for some of our other container deployments.

And more generally, here's what their current Dockerfile template looks like:
```
FROM registry.access.redhat.com/ubi7/ubi-minimal:latest
ENV OPERATOR=/usr/local/bin/{{.ProjectName}} \
    USER_UID=1001 \
    USER_NAME={{.ProjectName}}
COPY build/_output/bin/{{.ProjectName}} ${OPERATOR}
COPY build/bin /usr/local/bin
RUN  /usr/local/bin/user_setup
ENTRYPOINT ["/usr/local/bin/entrypoint"]
USER ${USER_UID}
```

Some of that (the user setup) seems moot as long as we continue to run as root. (Which we should try to get away from, and I have an idea about that, but that's a separate issue.) Other than that it APPEARS that the only interesting change is that base image. Concur?

Reply:
Will we get into subscription issues, if we have to do install any other software?

Reply:
I ran into issues trying to use the ubi-minimal image. I'm on CentOS and needed certificates to access the image. I had trouble generating the certificates so I fell back to using the alpine image.

Just let me know what image you guys decide on and I'll get it to work.

Reply:
Interesting. I just tried this command manually and it worked for me on my Mac laptop:
```shell
docker pull registry.access.redhat.com/ubi7/ubi-minimal:latest
```

What step of the process was failing for you, and what was the error?

(I mean Alpine is probably fine but I can imagine we might get dinged for not using UBI, so I'm interested in seeing if that's a possibility for us or not.)

Reply:
Sorry, got distracted by a different bug. Anyway it fails for me too on CentOS, but not in any way related to certs (apparently):
```shell
[sources]$ docker pull registry.access.redhat.com/ubi7/ubi-minimal:latest
latest: Pulling from registry.access.redhat.com/ubi7/ubi-minimal
b54017d3c94f: Pulling fs layer
b526fa5b30f2: Pulling fs layer
Pulling repository registry.access.redhat.com/ubi7/ubi-minimal
Error: image ubi7/ubi-minimal:latest not found
```

I did come across this other bug report which does mention certs and seems to conclude "Docker version too old": https://github.com/operator-framework/operator-sdk/issues/1486

FWIW the Docker version in my CentOS VM is super old (1.7.1) while my Mac version is fairly recent (Docker Engine Community 18.09.2). What Docker version are you rolling with?

Reply:
Here's the error I'm seeing:

Sending build context to Docker daemon 320.8 MB
Step 1/6 : FROM registry.access.redhat.com/ubi7/ubi-minimal:latest
Trying to pull repository registry.access.redhat.com/ubi7/ubi-minimal ...
open /etc/docker/certs.d/registry.access.redhat.com/redhat-ca.crt: no such file or directory

Reply:
Yeah that seems to be the "docker version too old" thing mentioned above. What Docker version do you have?

Reply:
docker-client-1.13.1-96.gitb2f74b2.el7.centos.x86_64
docker-common-1.13.1-96.gitb2f74b2.el7.centos.x86_64
docker-1.13.1-96.gitb2f74b2.el7.centos.x86_64

Reply:
OK so FWIW, here's how I got that to work.

Those certificates are supposed to live in a certain folder. That folder is supplied by the python-rhsm-certificates package. However on my test host (EC2 VM), it's also supplied by the currently installed subscription-manager-rhsm-certificates package, which just puts an empty directory there. That package is marked as obsoleting python-rhsm-certificates.

So I tried this:
```shell
rpm -e --nodeps subscription-manager-rhsm-certificates
yum --setopt=obsoletes=0 install python-rhsm-certificates
```
Then I could do the docker pull.

(I had to use --nodeps so that dependent packages like Docker wouldn't get yanked out.)

This is not a super great solution but maybe it helps point in the right direction? And if your current CentOS environment does NOT have the subscription-manager-rhsm-certificates package installed, then this may be a super easy fix and all you need to do is that yum install command above.

Reply:
Your workaround worked for me. I tested with ubi-minimal as the base image and it seems to work ok.